examples/hpc/dla-driver-ffi/src/lib.rs

```diff
@@ -28,7 +29,7 @@ unsafe fn ffi_data_import(
     kernel_width: usize,
     kernel_order: *const c_char,
 ) -> (Tensor3<i8>, Tensor4<i8>) {
```

Comment:
You do have to wonder if some of these could or should be combined into structs even for the FFI, e.g.,

```c
typedef struct data_t {
  int8_t *data,
  size_t channels,
  size_t height,
  size_t width,
  char *order,
} data_t;`
```

```c
data_t input { .data = ... }
...
ffi_data_import(input, kernel)
```

Might not make sense to commit to refactoring at this point in the project though unless we have some reason to continue the work next year :)

Reply:
Yes, if I or someone else want's to come back to this for optimizations at some point, this should be done. Hard to justify now.